fenix/app/src/test/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckServiceImplTest.kt

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+import kotlinx.coroutines.test.runTest
+import mozilla.components.browser.state.state.BrowserState
+import mozilla.components.browser.state.state.createTab
+import mozilla.components.browser.state.store.BrowserStore
+import mozilla.components.concept.engine.EngineSession
+import mozilla.components.concept.engine.shopping.ProductAnalysis
+import mozilla.components.support.test.any
+import mozilla.components.support.test.mock
+import mozilla.components.support.test.rule.MainCoroutineRule
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito
+
+class ReviewQualityCheckServiceImplTest {
+
+    @get:Rule
+    val coroutinesTestRule = MainCoroutineRule()
+
+    @Test
+    fun `GIVEN fetch is called WHEN onResult is invoked THEN product analysis returns the same data`() =
```

Comment:
given the test WHEN fetch is called THEN fetch is called for the selected tab I wonder how useful this is

Reply:
So seems like you're referencing the 3rd test in this file, basically it's testing that we are infact using the selected tab url, if not the request would fail or worse, show a different product analysis. So now if I change the implementation to `firstTab` or something else, this test will catch it.